service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -231,28 +222,23 @@ private boolean deleteResourceInner(UUID workspaceId, UUID resourceId) {
     // There are three cases for the stewardship type filter
     // 1. If it is REFERENCED, then we ignore id list and just filter
     //    for referenced resources.
-    // 2. If it is CONTROLLED, and the id list is not empty, then we filter for
-    //    CONTROLLED and require that the resources be in the id list.
+    // 2. If it is CONTROLLED, then we filter for
+    //    CONTROLLED resources.
     // 3. If no filter is specified (it is null), then we want both REFERENCED
     //    and CONTROLLED resources; that is, we want the OR of 1 and 2
     boolean includeReferenced = (stewardshipType == null || stewardshipType == REFERENCED);
-    boolean includeControlled =
-        ((controlledResourceIds != null && !controlledResourceIds.isEmpty())
-            && (stewardshipType == null || (stewardshipType == CONTROLLED)));
+    boolean includeControlled = (stewardshipType == null || stewardshipType == CONTROLLED);
 
     final String referencedPhrase = "stewardship_type = :referenced_resource";
-    final String controlledPhrase =
-        "(stewardship_type = :controlled_resource AND resource_id IN (:id_list))";
+    final String controlledPhrase = "stewardship_type = :controlled_resource";
```

Comment:
What does the colon do here?

Reply:
It's a query parameter referring to the value `CONTROLLED.toSql()` that we put in the map earlier. I like that a bit more than string concatenation here, but it's mostly style/convenience

Reply:
Oh right, it's a replacement token.